src/charm.py

```diff
@@ -111,7 +104,12 @@ def change_config(self) -> None:
         except PebbleServiceError as exc:
             self.model.unit.status = ops.BlockedStatus(str(exc))
             return
-        self.replan_nginx()
+        container = self.unit.get_container(synapse.SYNAPSE_NGINX_CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
```

Comment:
Should we maybe say which pebble we're waiting for here? Same comment for others below.

Reply:
Sure, changed.

Reply:
I'll close this one but feel free to re-open it.